### PR TITLE
feat: [CI-14600]: Conversion Spec for readMavenPom Jenkins Step

### PR DESCRIPTION
**Pull request by Ompragash**
Added "readMavenPom" support to go-convert specific to jenkins migration.

ReadMavenPom Jenkins Trace: [op-readMavenPom__1.json](https://github.com/user-attachments/files/17281424/op-readMavenPom__1.json)

Test Output:
```bash
go run main.go jenkinsjson -downgrade op-readMavenPom__1.json
```

<details>
  <summary>Example Harness v0 YAML</summary>

```yaml
pipeline:
  identifier: default
  name: default
  orgIdentifier: default
  projectIdentifier: default
  properties:
    ci:
      codebase:
        build: <+input>
  stages:
  - stage:
      identifier: build
      name: build
      spec:
        cloneCodebase: true
        execution:
          steps:
          - stepGroup:
              identifier: Stage__nullaa0159
              name: Download Sample POM
              steps:
              - step:
                  identifier: sh83952b
                  name: sh
                  spec:
                    command: "\n                    curl -o pom.xml https://raw.githubusercontent.com/spring-projects/spring-petclinic/main/pom.xml\n
                      \               "
                    image: alpine
                    shell: Sh
                  timeout: ""
                  type: Run
              timeout: ""
          - stepGroup:
              identifier: Stage__null871e47
              name: Read Maven POM
              steps:
              - step:
                  identifier: readmavenpomf2e306
                  name: readMavenPom
                  spec:
                    image: harnesscommunity/drone-get-maven-version:latest
                    settings:
                      pom_path: pom.xml
                  timeout: ""
                  type: Plugin
              timeout: ""
        platform:
          arch: Amd64
          os: Linux
        runtime:
          spec: {}
          type: Cloud
      type: CI
```

</details>

<details>
  <summary>Example Harness v1 YAML</summary>

```yaml
version: 1
kind: pipeline
type: ""
name: ""
spec:
  stages:
  - desc: ""
    id: build
    name: build
    strategy: null
    delegate: []
    status: null
    type: ci
    when: null
    failure: null
    inputs: {}
    spec:
      cache: null
      clone: null
      platform: null
      runtime: null
      steps:
      - id: Stage__nullaa0159
        name: Download Sample POM
        desc: ""
        type: group
        timeout: ""
        strategy: null
        when: null
        failure: null
        inputs: {}
        spec:
          steps:
          - id: sh83952b
            name: sh
            desc: ""
            type: script
            timeout: ""
            strategy: null
            when: null
            failure: null
            inputs: {}
            spec:
              image: alpine
              connector: ""
              user: ""
              group: ""
              pull: ""
              shell: sh
              envs: {}
              run: "\n                    curl -o pom.xml https://raw.githubusercontent.com/spring-projects/spring-petclinic/main/pom.xml\n
                \               "
              entrypoint: ""
              args: []
              privileged: false
              network: ""
              reports: []
              outputs: []
              resources: null
              mount: []
      - id: Stage__null871e47
        name: Read Maven POM
        desc: ""
        type: group
        timeout: ""
        strategy: null
        when: null
        failure: null
        inputs: {}
        spec:
          steps:
          - id: readMavenPomf2e306
            name: readMavenPom
            desc: ""
            type: plugin
            timeout: ""
            strategy: null
            when: null
            failure: null
            inputs: {}
            spec:
              image: harnesscommunity/drone-get-maven-version:latest
              name: ""
              uses: ""
              connector: ""
              pull: ""
              envs: {}
              reports: []
              privileged: false
              user: ""
              group: ""
              network: ""
              with:
                pom_path: pom.xml
              inputs: {}
              outputs: []
              resources: null
              mount: []
      envs: {}
      volumes: []
  inputs: {}
  options: null
```

</details>

### CODE REVIEW

File: samples/jenkins/Jenkinsfile [hunk 1]
```diff
@@ -143,5 +143,26 @@ line3'''
...
+        }
+
+        // Reading the downloaded POM file
+        stage('Read Maven POM') {
+            steps {
+                script {
+                    def pom = readMavenPom file: 'pom.xml'
+
+                    // Extracting the version from the POM file
+                    echo "Project Version: ${pom.version}"
```
[the remark this answers is not in the document]
Comment (Ompragash) [reply samples/jenkins/Jenkinsfile:163]: currently it only extracts version from pom.xml file

File: convert/jenkinsjson/convertTestFiles/readMavenPom/readMavenPom.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+      - step:
+          identifier: readmavenpomf2e306
+          name: readMavenPom
+          spec:
+            image: harnesscommunity/drone-get-maven-version:latest
+            settings:
+              pom_path: pom.xml
+          timeout: ""
+          type: Plugin
+    timeout: ""
```
[the remark this answers is not in the document]
Comment (Ompragash) [reply convert/jenkinsjson/convertTestFiles/readMavenPom/readMavenPom.yml:14]: output `POM_VERSION` and can be used in the subsequent steps with `<+steps.STEP_ID.output.outputVariables.POM_VERSION>`

File: convert/jenkinsjson/json/readMavenPom.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package json
+
+import (
+	harness "github.com/drone/spec/dist/go"
+)
+
+func ConvertReadMavenPom(node Node) *harness.Step {
+
+	settings := map[string]interface{}{
+		"pom_path": node.ParameterMap["file"],
```
Comment (hemanthmantri) [inline convert/jenkinsjson/json/readMavenPom.go:10]: Is this always a single entry and not a list of files?

Comment (jamie-harness) [reply convert/jenkinsjson/json/readMavenPom.go:10]: https://www.jenkins.io/doc/pipeline/steps/pipeline-utility-steps/#readmavenpom-read-a-maven-project-file Single entry, exact file

**Pull request merged**